### PR TITLE
Expose `Credentials::ConvertX509CertToChipCert` to Obj-C.

### PR DESCRIPTION
**Pull request by mburshteyn1**
#### Problem
* Expose `Credentials::ConvertX509CertToChipCert` to Obj-C.

#### Change overview
Added function `convertToCHIPCertFromX509Cert` to `MTRCertificates` to expose `Credentials::ConvertX509CertToChipCert` to Obj-C.

### CODE REVIEW

**Review by emargolis — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRCertificates.mm [hunk 1]
```diff
@@ -196,4 +197,21 @@ + (nullable NSData *)generateCertificateSigningRequest:(id<MTRKeypair>)keypair
     return nil;
 }
 
++ (nullable NSData *)convertToCHIPCertFromX509Cert:(NSData *)x509Certificate
+{
+
+    chip::ByteSpan x509CertBytes = chip::ByteSpan((uint8_t *) x509Certificate.bytes, x509Certificate.length);
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRCertificates.mm:203]: Please use AsByteSpan().

File: src/darwin/Framework/CHIP/MTRCertificates.mm [hunk 2]
```diff
@@ -196,4 +197,21 @@ + (nullable NSData *)generateCertificateSigningRequest:(id<MTRKeypair>)keypair
...
+
+    NSMutableData * chipCertBuffer = [[NSMutableData alloc] initWithLength:chip::Credentials::kMaxCHIPCertLength];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRCertificates.mm:205]: Please use a stack buffer.

File: src/darwin/Framework/CHIP/MTRCertificates.mm [hunk 3]
```diff
@@ -196,4 +197,21 @@ + (nullable NSData *)generateCertificateSigningRequest:(id<MTRKeypair>)keypair
...
+    chip::MutableByteSpan chipCertBytes((uint8_t *) chipCertBuffer.mutableBytes, chip::Credentials::kMaxCHIPCertLength);
+
+    CHIP_ERROR errorCode = chip::Credentials::ConvertX509CertToChipCert(x509CertBytes, chipCertBytes);
+    MTR_LOG_ERROR("ConvertX509CertToChipCert: %{public}s", chip::ErrorStr(errorCode));
+
+    if (errorCode != CHIP_NO_ERROR)
+        return nil;
+
+    return [NSData dataWithBytes:chipCertBytes.data() length:chipCertBytes.size()];
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRCertificates.mm:214]: Please use `AsData`.

File: src/darwin/Framework/CHIP/MTRCertificates.h [hunk 4]
```diff
@@ -125,6 +125,9 @@ NS_ASSUME_NONNULL_BEGIN
 + (nullable NSData *)generateCertificateSigningRequest:(id<MTRKeypair>)keypair
                                                  error:(NSError * __autoreleasing _Nullable * _Nullable)error;
 
+/** Converts the given X.509v3 certificate to the CHIP certificate format. */
++ (nullable NSData *)convertToCHIPCertFromX509Cert:(NSData *)x509Certificate;
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTRCertificates.h:129]: Needs to document when null is returned.

**Pull request merged**